plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -704,7 +704,7 @@ public static function current_screen( $screen ) {
 	 * Initiate a new OAuth connection.
 	 */
 	private static function _helper_auth_connect() {
-		if ( empty( $_GET['wc-helper-nonce'] ) || ! wp_verify_nonce( wp_unslash( $_GET['wc-helper-nonce'] ), 'connect' ) ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		if ( empty( $_GET['wc-is-cli'] ) && ( empty( $_GET['wc-helper-nonce'] ) || ! wp_verify_nonce( wp_unslash( $_GET['wc-helper-nonce'] ), 'connect' ) ) ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```

Comment:
nonce generated in CLI can not be verified using `wp_verify_nonce` since it depends on the session token.